lib/request_handler/validation/backend.rb

```diff
@@ -0,0 +1,42 @@
+
+require_relative 'errors'
+
+module RequestHandler
+  module Validation
+    class Backend
+      def self.valid_schema?
+        raise NotImplementedError
+      end
+
+      def initialize(value, schema, options: {})
```

Comment:
Is it actually necessary for the usage of the backends that they are stateful and make creating a new instance necessary - with just dry-validation that was not necessary IIRC

Reply:
the schema parse works with the result of the validation, and that one needs have the same interface accross all backends, that why I did it like this

Reply:
the how about wrapping the result of both engines in a "result" object ?

Reply:
yeah that would work, I dont see much benefit over the current solution though?